### PR TITLE
Heading control

### PR DESCRIPTION
**Pull request by AidanSmith**
Head should stay straight while strafing, using PID. I and D are not calibrated right now.

### CODE REVIEW

**Comment by AidanSmith** [comment]
I fixed a few small bugs that I found.

File: QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java [hunk 1]
```diff
@@ -100,10 +115,29 @@ public void event() {
 	private static EventOutput calibrate = new EventOutput() {
 		public void event() {
 			float yaw = HeadingSensor.yaw.get();
+			desiredAngle.set(yaw);
+			if (Igneous.getIsDisabled().get()){
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java]: You should stick `Igneous.getIsDisable()` into a variable - you don't want it to be creating a new channel for you every single time.

File: QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java [hunk 2]
```diff
@@ -113,6 +147,19 @@ public static void setup() {
 		Cluck.publish("Zero Gyro", HeadingSensor.zeroGyro);
 		recalibrateButton.send(calibrate);
 		
+		FloatStatus p = context.getFloat("drive-p", 0.01f);
+		FloatStatus i = context.getFloat("drive-i", 0f);
+		FloatStatus d = context.getFloat("drive-d", 0f);
+
+		pid = new PIDControl(adjustedYaw, desiredAngle, p, i, d);
+		pid.setOutputBounds(-1f, 1f);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java:155]: Consider putting bounds on the integral accumulator.

**Comment by celskeggs** [comment]
That's all I see right now.

**Comment by AidanSmith** [comment]
I made the requested changes. Does (-.5, .5) sound good for integral bounds?

**Comment by celskeggs** [comment]
Looks good. Have you tested this in the emulator?

**Comment by AidanSmith** [comment]
Does the heading sensor work on the emulator yet?

**Comment by celskeggs** [comment]
Nope. But the robot doesn't have one either yet, so I just need it tested without it.

**Comment by AidanSmith** [comment]
I tested it. I temporarily changed the heading sensor yaw to an encoder because in terms of the emulator it has similar functionality. I will now go on the resolve merge conflicts.

**Comment by AidanSmith** [comment]
I fixed the conflict and changed yaw back to the heading sensor.

**Comment by celskeggs** [comment]
Excellent.

**Pull request merged**